lib/Watcher.php

```diff
@@ -172,7 +172,17 @@ public function postWrite(Node $node) {
 			// note that invalidatePersons depends on existence of faces for a given image,
 			// and we must invalidate before we delete faces!
 			$this->personMapper->invalidatePersons($imageId);
+
+			// Fetch all faces to be deleted before deleting them, and then delete them
+			$facesToRemove = $this->faceMapper->findByImage($imageId);
 			$this->faceMapper->removeFaces($imageId);
+
+			// If any person is now without faces, remove those (empty) persons
+			foreach ($facesToRemove as $faceToRemove) {
+				if ($faceToRemove->getPerson() !== null) {
```

Comment:
In that circumstance this would Null?. While in the background task the face is discovered and grouped?

Reply:
well, what if you add image, do face detection and there is no clustering yet (still not enough images...). One way to reach this situation!:)